src/viam/app/data_client.py

```diff
@@ -60,6 +62,54 @@ class DataClient:
     `ViamClient`.
     """
 
+    class TabularData:
+        """Class representing a piece of tabular data and associated metadata.
+
+        Args:
+            data (Mapping[str, Any]): the requested data.
+            metadata (viam.proto.app.data.CaptureMetadata): the metadata from the request.
+            time_requested (datetime): the time the data request was sent.
+            time_received (datetime): the time the requested data was received.
+        """
+
+        def __init__(self, data: Mapping[str, Any], metadata: CaptureMetadata, time_requested: datetime, time_received: datetime) -> None:
```

Comment:
Do we need to annotate the init function's return type? If we do, should it be "Self" or "TabularData" instead of "None"?

Reply:
Yeah, good question! I don't know that we have to annotate, we seem to be a bit inconsistent about it in the SDK. Personally I'm a fan of type annotations so I'd prefer to keep it in, and when we do annotate `init` methods in python it seems to pretty consistently set the return type as `None`.